### PR TITLE
BUGFIX: Graceful error handling if setup not completed

### LINKED ISSUE #4446: BUG: SiteDetection fails if no site exists
Neos totally breaks if no site is set. Not even other flow routes work

image

when the middle ware would just be okay with no site, it correctly only fails for probable neos pages:

image

or 

> Exception in line 513 of /Flow_Object_Classes/Neos_Neos_Controller_Frontend_NodeController.php: TODO: SITE NOT FOUND; should not happen (for address NodeAddress[contentStream=62c42194-8d0b-4835-80ab-b8032968cc5a, dimensionSpacePoint={"language":"en_US"}, nodeAggregateId=66807a8d-1914-4904-b531-309b3b241c2f, workspaceName=live]

expected:

routing works for normal flow routes, and other middlewares work

screen for unmatched routes (where it might be a neos page)

image

### LINKED ISSUE #4714: Neos 9 `NoHomepageException` is never thrown
Related https://github.com/neos/neos-development-collection/issues/4446

In Neos 8.3 we are used to this screen:

> Homepage could not be loaded. Probably you haven't imported a site yet

but in neos 9 we never throw this exception and allow the path to be covered by another route:

image

This is way more correct as throwing the `NoHomepageException` whenever no node matched and the request uri path is `=== ""` empty

But for example the generic `NoHomepageException` might be misleading for these cases:

- when there is no root path handler configured and only `/en_US` works.
- when there is no site configured for a domain in a multi site setup and thus no document node matches

Question is how to continue here? Do we want a route with a super low priority which just tells that there is something wrong in the Neos.Neos usecase? I think its not the purpose of the `EventSourcedFrontendNodeRoutePartHandler`

### PR DESCRIPTION
**Pull request by mhsdesign**
Resolves: https://github.com/neos/neos-development-collection/issues/4714
Resolves: https://github.com/neos/neos-development-collection/issues/4446

Current when visiting `http://127.0.0.1:8081` or `http://127.0.0.1:8081/anythingunmatched` without a neos site being imported:

(yes a white-screen if a site exists but no site node)
image

(or if you have not site the `EventSourcedFrontendNodeRoutePartHandler`)
image

Proposed:

image

As soon as a site exist, the fusion exception view will use the `error` path as usual to render your custom 404 page.

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by mhsdesign** [comment]
@kitsunet that should fix most cases and fix the flow tests, ~but im not sure this is all perfect. Seems a bit hacked together, and works only well in exactly this alignment~

With https://github.com/neos/neos-development-collection/pull/4880 its nearly perfect :)

**Comment by mhsdesign** [comment]
This pr makes https://github.com/neos/neos-development-collection/pull/4050 obsolete.

**Comment by mhsdesign** [comment]
The Flow functional tests pass now when Neos is installed. So under that measurement the change is correct :D

File: Neos.Neos/Classes/View/FusionExceptionView.php [hunk 1]
```diff
@@ -91,7 +92,7 @@ class FusionExceptionView extends AbstractView
     protected DomainRepository $domainRepository;
 
     /**
-     * @return string
+     * @return mixed
```
Comment (bwaidelich) [inline Neos.Neos/Classes/View/FusionExceptionView.php:95]: nitpick: is this really `mixed` or can we have a more specific return type`

**Review by bwaidelich — APPROVED** [review]
Nice, +1 by reading

File: Neos.Neos/Classes/View/FusionExceptionView.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.Neos/Classes/View/FusionExceptionView.php:95]: yeah ... i fiddled around, truth is the `FusionView` for welcomeTemplate returns `mixed` thats why

File: Neos.Neos/Configuration/Settings.yaml [hunk 2]
```diff
@@ -481,7 +481,7 @@ Neos:
 
           noHomepageException:
             matchingExceptionClassNames:
-              - Neos\Neos\Routing\Exception\NoHomepageException
+              - Neos\Neos\FrontendRouting\SiteDetection\SiteDetectionFailedException
             options: *welcomeTemplate
```
Comment (mhsdesign) [inline Neos.Neos/Configuration/Settings.yaml]: i have to evaluate if a `SiteDetectionFailedException` will with my adjustments even come that far ;)

Comment (mhsdesign) [reply Neos.Neos/Configuration/Settings.yaml]: Yes the only case where this currently happens, is at the `neos/content` route, as the backend controller cannot find a site.

But in 8.3 we dont have a graceful error as well only something like `Site doest not exist for current user`: https://github.com/neos/neos-ui/issues/3182

We should probably redirect to some other available module, or at least keep the backend frame around and not destroy everything.

Comment (mhsdesign) [reply Neos.Neos/Configuration/Settings.yaml]: i will remove the `noHomepageException` rendering group as this should be solve differently see https://github.com/neos/neos-ui/issues/3721

**Pull request merged**